app.rb

```diff
@@ -29,8 +29,28 @@ class App < Sinatra::Base
   end
 
   post '/user-signup/email-notification' do
-    logger.info(request)
-    WifiUser::UseCase::SnsNotificationHandler.new(logger).handle(request)
+    whitelist_checker = WifiUser::UseCases::CheckIfWhitelistedEmail.new(
```

Comment:
I think these can be lifted even higher, and moved to the app initialisation phase instead of being part of the controller.

The only thing tied to the controller is the final execution.
Feel it will make controller testing easier (app configuration can be modified instead of patching the classes), and also improve efficiency (only initialised once)

Reply:
Given this is a refactoring, I'm not making any improvements just yet.
Happy to have a discussion, also have some ideas around this.

Reply:
sure thing